src/app/show.service.ts

```diff
@@ -1,9 +1,42 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { IShowDetailsData } from './ishow-details-data'
+import { IShowDetails } from './ishow-details';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ShowService {
 
-  constructor() { }
+  constructor(private httpClient: HttpClient) {}
+
+  getShow(showId: string){
+    return this.httpClient.get<IShowDetailsData>(`http://api.tvmaze.com/shows/${showId}?embed=cast`)
+    }
+  
+  transformToIShowDetails(data:IShowDetailsData):IShowDetails{
+    return {
+      showTitle: data.name,
+      showImage: data.image.medium,
+      showLink: data.officialSite,
+      rating: data.rating.average,
+      country: data.network.country,
+      channel: data.network.name,
+      genre: data.genres.toString(),
+      language: data.language,
+      previousEpisode: data._links.previousepisode.href,
+      premierDate: new Date (data.premiered),
+      summary: data.summary,
+      schedule:{
+        days: data.schedule.days.toString(),
+        time: new Date (data.schedule.time)
+      },
+      cast:{
+        castName: data._embedded.cast[0].person.name,
```

Comment:
I filled this in as being for item 0 in the array... but I am trying to envision a way to loop this to gather items 1, 2, and 3 as well so there can be four cast member 'tiles' on the show page.  Would love any ideas!